File: app/src/main/java/com/scatl/uestcbbs/custom/imageview/ArcImageView.java
package com.scatl.uestcbbs.custom.imageview;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Canvas;
import android.graphics.Path;
import android.util.AttributeSet;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatImageView;

import com.scatl.uestcbbs.R;


public class ArcImageView extends AppCompatImageView {

    private Path path;
    private int arcHeight;

    public ArcImageView(Context context) {
        super(context);
        init();
    }

    public ArcImageView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init();

        TypedArray typedValue = context.obtainStyledAttributes(attrs, R.styleable.ArcImageView);
        arcHeight = typedValue.getInt(R.styleable.ArcImageView_arcHeight, 50);
        typedValue.recycle();
    }

    public ArcImageView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        TypedArray typedValue = context.obtainStyledAttributes(attrs, R.styleable.ArcImageView);
        arcHeight = typedValue.getInt(R.styleable.ArcImageView_arcHeight, 50);
        typedValue.recycle();
    }

    private void init() {
        path = new Path();
    }

    @Override
    protected void onDraw(Canvas canvas) {

        //path.moveTo(0, 0);
        //path.lineTo(0, getHeight());
        //path.quadTo(getWidth() / 2, getHeight()-50, getWidth(), getHeight());
        //path.lineTo(getWidth(), 0);


        path.moveTo(0, getHeight() - arcHeight);
        path.quadTo(getWidth() / 2, getHeight(), getWidth(), getHeight() - arcHeight);
        path.lineTo(getWidth(), getHeight() - arcHeight);
        path.lineTo(getWidth(), 0);
        path.lineTo(0, 0);
        path.close();
        canvas.clipPath(path);
        //canvas.setDrawFilter(new PaintFlagsDrawFilter(0, Paint.FILTER_BITMAP_FLAG|Paint.ANTI_ALIAS_FLAG));
        super.onDraw(canvas);

    }

}

File: widget/src/main/java/com/scatl/widget/imageview/ArcImageView.kt
package com.scatl.widget.imageview

import android.content.Context
import android.graphics.Canvas
import android.graphics.Path
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView
import com.scatl.widget.R

/**
 * Created by sca_tl at 2023/5/26 15:04
 */
class ArcImageView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) : AppCompatImageView(context, attrs) {

    private var path: Path? = null

    var arcHeight = 50F
        set(value) {
            field = value
            invalidate()
        }

    init {
        if (attrs != null) {
            val typedArray = context.obtainStyledAttributes(attrs, R.styleable.ArcImageView)
            arcHeight = typedArray.getDimension(R.styleable.ArcImageView_arcHeight, 50F)
            typedArray.recycle()
        }
        path = Path()
    }

    override fun onDraw(canvas: Canvas) {
        path?.moveTo(0f, height - arcHeight)
        path?.quadTo(width / 2f, height.toFloat(), width.toFloat(), height - arcHeight)
        path?.lineTo(width.toFloat(), height - arcHeight)
        path?.lineTo(width.toFloat(), 0f)
        path?.lineTo(0f, 0f)
        path?.close()
        canvas.clipPath(path!!)
        super.onDraw(canvas)
    }
}